modules/javafx.graphics/src/main/java/javafx/scene/robot/Robot.java

```diff
@@ -0,0 +1,373 @@
+     * screen.
+     * @return the screen capture of the specified {@code region} as a {@link WritableImage}
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public WritableImage getScreenCapture(WritableImage image, double x, double y,
+                                          double width, double height, boolean scaleToFit) {
+        return peer.getScreenCapture(image, x, y, width, height, scaleToFit);
+    }
+
+    /**
+     * Returns a {@link WritableImage} containing the specified rectangular area of the
+     * primary screen. If the given {@code image} is {@literal null}, or if the given
+     * {@code image} is too small, a new {@code WritableImage} will be created and returned.
+     * Otherwise, the given {@code image} is re-used.
+     *
+     * @implSpec This method is equivalent to calling {@code getScreenCapture(x, y, width, height, true)},
+     * that is, this method scales the image to fit the requested size.
+     * @param image either {@literal null} or a {@code WritableImage} that will
+     * be used to place the screen capture in
+     * @param x the starting x-position of the rectangular area to capture
+     * @param y the starting y-position of the rectangular area to capture
+     * @param width the width of the rectangular area to capture
+     * @param height the height of the rectangular area to capture
+     * @return the screen capture of the specified {@code region} as a {@link WritableImage}
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public WritableImage getScreenCapture(WritableImage image, double x, double y,
+                                          double width, double height) {
+        return getScreenCapture(image, x, y, width, height, true);
+    }
+
+    /**
+     * Returns a {@link WritableImage} containing the specified rectangular area of the
+     * primary screen. If the given {@code image} is {@literal null}, or if the given
+     * {@code image} is too small, a new {@code WritableImage} will be created and returned.
+     * Otherwise, the given {@code image} is re-used.
+     *
+     * @implSpec This method is equivalent to calling {@code getScreenCapture(image, region, true)},
+     * that is, this method scales the image to fit the requested size.
+     * @param image either {@literal null} or a {@code WritableImage} that will
+     * be used to place the screen capture in
+     * @param region the rectangular area of the screen to capture
+     * @return the screen capture of the specified {@code region} as a {@link WritableImage}
+     * @throws IllegalStateException if this method is called on a thread
+     * other than the JavaFX Application Thread.
+     */
+    public WritableImage getScreenCapture(WritableImage image, Rectangle2D region) {
+        Objects.requireNonNull(region);
```

Comment:
npe if image/region is null?

Reply:
Yes for region, no for image (as explained above).